### ISSUE TITLE
No particle size update during merging with MERCURIUS integrator

### ISSUE THREAD

**Issue by Astronnet** [hannorein/rebound#322]
Hi ! I am using the MERCURIUS integrator with direct collision search and merging and noticed that the particles.r is not updated after a merging occurs. The mass/velocity of the particle is updated though. When I run the same integration with IAS15 instead, all the quantities are updated fine.

**Comment by hannorein** [comment]
Hm. Interesting. Could you provide me with a short piece of code that illustrates the problem?

**Comment by Astronnet** [comment]
Sure ! Here it is, 50 embryos with masses 0.05 Earth_mass randomly distributed from 0.7 to 1.0 AU integrated with MERCURIUS :
`int main(int argc, char* argv[]){
  struct reb_simulation* r = reb_create_simulation();
  double tmax  = 1.e7*2.*M_PI; 
  int N_emb      = 50;

  // Simulation Setup
  r->integrator	= REB_INTEGRATOR_MERCURIUS;
  r->heartbeat	= heartbeat;
  r->testparticle_type = 1;
  r->dt = 0.1;

  // Collisions
  r->collision = reb_simulation::REB_COLLISION_DIRECT;
  r->collision_resolve = reb_collision_resolve_merge;
  r->collision_resolve_keep_sorted = 1;

  // Sun
  struct reb_particle star = {0};
  star.m = 1.;
  star.r = 0.1;
  reb_add(r, star);
	
  // Jupiter
  /*struct reb_particle jup = {0};
  jup.m = 1.e-3;
  jup   = reb_tools_orbit_to_particle(r->G, star, jup.m, 5.4, 0., 0., 0., 0., 0.);
  jup.r = 1.4e8/1.49e11;
  reb_add(r, jup);*/
	
  //Saturn
  /*struct reb_particle sat = {0};
  double pos = 5.4*pow(1.5,2./3.);
  sat.m = 5.68e26/1.89e30;
  sat   = reb_tools_orbit_to_particle(r->G, star, sat.m, pos, 0., 0., 0., 0., 0.);
  sat.r = 1.4e8/1.49e11;
  reb_add(r, sat);*/
	
		
  // Embryos (massive particles)
  double a_min  = 0.7;
  double a_max  = 1.0;
	
	
  for(int i=0;i<N_emb;i++){
    struct reb_particle pt = {0};
    double a     = reb_random_uniform(a_min,a_max);
    double e     = reb_random_rayleigh(0.0005);
    double inc   = reb_random_rayleigh(0.0002);
    double Omega = reb_random_uniform(0,2.*M_PI);
    double apsis = reb_random_uniform(0,2.*M_PI);
    double phi 	 = reb_random_uniform(0,2.*M_PI);
    pt.m         = 0.05*6.e24/1.89e30;
    pt           = reb_tools_orbit_to_particle(r->G, star, pt.m, a, e, inc, Omega, apsis, phi);
    pt.r         = pow(.75*pt.m*1.89e30/M_PI/4.e3,1./3.)/1.49e11;   
    reb_add(r, pt);
  }
  r->N_active = r->N;

  reb_integrate(r, tmax);
	
}`

**Comment by Astronnet** [comment]
Sorry, some of the '*' characters are missing. The part where I add Jupiter and Saturn should be commented, it is not necessary to exemplify the issue. Also, never mind the "r->collision = reb_simulation::REB_COLLISION_DIRECT", it should simply be "r->collision = REB_COLLISION_DIRECT". I hope this helps !

**Comment by hannorein** [comment]
I'm sorry, but there are so many typos in the above code! I don't want to change your code too much as I might accidentally fix the problem. Could you try to copy and paste this again? Thanks!

**Comment by Astronnet** [comment]
Yes sorry, I don't know what happened, I tried to copy/paste that in a code section but it went wrong (I'm new to github) ! This should be better, please let me know if you need anything else.

> int main(int argc, char* argv[]){
>   struct reb_simulation* r = reb_create_simulation();
>   double tmax  = 1.e7*2.*M_PI;
>   int N_emb      = 50;
> 
>   // Simulation Setup
>   r->integrator	= REB_INTEGRATOR_MERCURIUS;
>   r->heartbeat	= heartbeat;
>   r->dt = 0.1;
> 
>   // Collisions
>   r->collision = REB_COLLISION_DIRECT;
>   r->collision_resolve = reb_collision_resolve_merge;
>   r->collision_resolve_keep_sorted = 1;
> 
>   // Sun
>   struct reb_particle star = {0};
>   star.m = 1.;
>   star.r = 0.1;
>   reb_add(r, star);
> 		
>   // Embryos (massive particles)
>   double a_min  = 0.7;
>   double a_max  = 1.0;
> 		
>   for(int i=0;i<N_emb;i++){
>     struct reb_particle pt = {0};
>     double a          = reb_random_uniform(a_min,a_max);
>     double e          = reb_random_rayleigh(0.0005);
>     double inc        = reb_random_rayleigh(0.0002);
>     double Omega = reb_random_uniform(0,2.*M_PI);
>     double apsis    = reb_random_uniform(0,2.*M_PI);
>     double phi       = reb_random_uniform(0,2.*M_PI);
>     pt.m = 0.05*6.e24/1.89e30;
>     pt   = reb_tools_orbit_to_particle(r->G, star, pt.m, a, e, inc, Omega, apsis, phi);
>     pt.r = pow(.75*pt.m*1.89e30/M_PI/4.e3,1./3.)/1.49e11;        
>     reb_add(r, pt);
>   }
> 
>   r->N_active = r->N;
>  
>   reb_integrate(r, tmax);
> 	
> }

**Comment by hannorein** [comment]
I've pushed a changed that should fix the radius issue.

There might be a related issue with the mass. Although I'm not sure at this point when there might be an issue. It is a bit more complicated than I originally hoped. It could depend on the coordinate system used.
